test/libyul/yulOptimizerTests/stackCompressor/unusedPrunerWithMSize.yul

```diff
@@ -0,0 +1,36 @@
+{
+    {
+        let _13_71 := 1
+        let _17_72 := pc()
+        let _22_75 := pc()
+        let _25_76 := pc()
+        let _30_80 := pc()
+        let _32_81 := pc()
+        // This should not be removed
+        pop(keccak256(1, 2))
```

Comment:
This test case passes on both unpatched and patched versions. The original test case correctly fails on the unpatched version and passes once this PR is applied. Please consider using the original test or adapting it such that we have a fail/pass transition pre/post PR.

Reply:
It does not pass for me. Are you sure?

Reply:
This actually proves that the test is correct: https://circleci.com/gh/ethereum/solidity/150576?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link